controller/src/lib.rs

```diff
@@ -102,32 +109,65 @@ pub async fn run_scheduler(nats: TypedNats, state: StateHandle) -> NeverResult {
                         })
                         .await?;
 
-                        ScheduleResponse::Scheduled {
-                            drone: drone_id,
-                            backend_id: spawn_request.backend_id,
-                            bearer_token: spawn_request.bearer_token.clone(),
-                            spawned: true,
-                        }
-                    }
-                    Ok(false) => {
-                        tracing::warn!("Drone rejected backend.");
-                        ScheduleResponse::NoDroneAvailable
-                    }
-                    Err(error) => {
-                        tracing::warn!(?error, "Scheduler returned error.");
-                        ScheduleResponse::NoDroneAvailable
+                    if let Some(lock) = schedule_request.value.lock.clone() {
+                        let notify = std::sync::Arc::new(Notify::new());
+                        lock_to_ready.insert(lock, notify.clone());
+                        tokio::spawn(async move {
+                            while state.state().get_logical_time() < seq_id {
```

Comment:
I don't like polling here, I implemented an alternative here: https://github.com/drifting-in-space/plane/pull/403